README.md

```diff
@@ -119,6 +123,55 @@ Serve files relative to `path`.
 Byte offset at which the stream starts, defaults to 0. The start is inclusive,
 meaning `start: 2` will include the 3rd byte in the stream.
 
+##### transform
+
+A function which returns a stream which can be used to transform the data.
+If this option is passed it will change the defaults of `etag` and `lastModified`
+to `false`.  If the transform stream changes the content length, it is also
+responsible for updating the `Content-Length` header or changing to
+`Transfer-Encoding: chunked`.
+
+The transform function is given the readable stream from `fs.createReadStream`,
+the `response` object and, the `options` passed to `send()`.  It must return
+a `ReadableStream`, you can even just return the `stream` passed to you.
+
+If you expect the transform to always return the same results, you can re-enable
+`etag` and `lastModified` and to use the underlying file as it normally would.
+
+*Note on range requests:* When a range request is sent, the read stream will
+only read part of the file. If your transform is preforming replacements, or
```

Comment:
This would violate the actual http spec, as the range is on the actual recieved output. This will be a large issue for example if the transform doubled the output (maybe encode as hex); the web browser will send a range way past the end of the actual underlying file.

Reply:
Do you think defaulting `acceptRanges: false` when `transform` is passed is a good compromise?

Reply:
If transform starts removing pretty much all the features of this module, at what point is the feature actually just a separate functionality?

Reply:
Well, default is not removal.  In my use case I actually would force set `etag` and `lastModified` to true and `acceptRanges` to false anyway.  To me that is not removing, just normal options setting for a specific use case.

Reply:
Sorry, to clarify i mean the removal as in what happens by default when the transform option is enabled. Yes, folks can then understand the foot guns and re enable certain things, but many will use the defaults.

Reply:
But I think the defaults are good mostly with the `transform` option, especially if we turn off range requests by default.  FWIW, range requests are so uncommon in my experience.  The only time I have ever used them was building a video player, but that is a very uncommon use case to optimize for.

If you are using the `transform` option, the default *should* be no range requests because it means you need to be very careful your transform can handle range requests.  The `content-length` is a little less clear, but it really depends on your transform, so it makes sense to not set it if you use `transform`.  Lastly for `last-modified`, this also depends on what your transform does so it makes sense to leave it up to the user.

The more I think about this, the more I think this is the right approach to a `transform` option default as long as it is well documented.

Reply:
I disagree on some of this; I think we can make a transform work with all the features if we implement it instead along the lines of a virtual fs, as I poked at in another comment, and not like this implementation. This implementation, ultimately, is just a simple solution that ends up having a lot of pitfalls I think don't need to be there if done differently. Range requests are not uncommon in practice; the module implements specifically what is not uncommon around the web. A very common point (besides audio and video files) that use range requests are pdf files too. That is how the web browser can display the first page of a 50mb pdf without waiting around forever. The browsers make use of range requests for quite a variety of file types under the hood.

Reply:
I also generally think it is bad API design for the default setting of one option to change based on what another, different option is set to, as it just adds confusion as to what exactly is going to happen with a set of options, as the defaults are now dynamic to other factors.